### PR TITLE
feat(iam): sample for receiving Cloud Audit Logs from Service Accout Key creation

### PR DESCRIPTION
**Pull request by muncus**
## Description

Sample demonstrates how to receive Cloud Audit Log events when a new key is
created for a service account.

## Checklist
- [x] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [ ] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [ ] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: eventarc/audit_iam/main.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+)
+
+func HandleCloudEvent(w http.ResponseWriter, r *http.Request) {
+	event, err := cloudevent.NewEventFromHTTPRequest(r)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintln(w, "Failed to create CloudEvent from request.")
+		log.Fatal("cloudevent.NewEventFromHTTPRequest:", err)
+	}
+	var logentry = auditdata.LogEntryData{}
```
Comment (telpirion) [inline eventarc/audit_iam/main.go:38]: suggestion: I typically initialize an empty struct with just a simple declaration.

```
var logEntry auditdata.LogEntryData
```

File: eventarc/audit_iam/main.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	// `protojson.Unmarshal`. UnmarshalOptions prevents this error.
+	umo := &protojson.UnmarshalOptions{DiscardUnknown: true}
+	err = umo.Unmarshal(event.Data(), &logentry)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintln(w, "Failed to parse Audit Log")
+		log.Fatal("protojson.Unmarshal:", err)
+	}
+	// Extract relevant fields from the audit log entry.
+	var auditlog = logentry.ProtoPayload
```
Comment (telpirion) [inline eventarc/audit_iam/main.go:49]: See previous about initializing an empty struct

File: eventarc/audit_iam/main.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	actor := auditlog.AuthenticationInfo.PrincipalEmail
+	principal := logentry.Resource.Labels["email_id"]
```
Comment (telpirion) [inline eventarc/audit_iam/main.go:51]: question: is "email_id" a defined & known key in the Labels map?

File: eventarc/audit_iam/main.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	// This key path can be used with gcloud to disable/delete the key:
+	// e.g. gcloud iam service-accounts keys disable ${keypath}
+	keypath := logentry.ProtoPayload.GetResponse().AsMap()["name"]
```
Comment (telpirion) [inline eventarc/audit_iam/main.go:54]: Same as above--is "name" a known key in this `map`? If so, I would include a link to the documentation that describes which keys are available.

File: eventarc/audit_iam/main_test.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	cloudevent "github.com/cloudevents/sdk-go/v2"
+	"github.com/googleapis/google-cloudevents-go/cloud/auditdata"
+	"github.com/googleapis/google-cloudevents-go/third_party/googleapis/google/api"
+)
+
+func TestIamAuditlogs(t *testing.T) {
```
Comment (telpirion) [inline eventarc/audit_iam/main_test.go:28]: I'm definitely going to bookmark this test--I always get stuck whenever I need to write a test that records an HTTP response.

File: eventarc/audit_iam/main.go
Comment (telpirion) [inline eventarc/audit_iam/main.go:1]: observation: there doesn't seem to be visual separation between comments and code. I would insert line breaks in between different logical chunks of code.

**Review by telpirion — CHANGES_REQUESTED** [review]

File: eventarc/audit_iam/main.go (hunk 3 above)
Comment (muncus) [reply eventarc/audit_iam/main.go]: I could not find any *documentation* about this map, so i've switched to the Resource Name here instead, which is pulled from the Request. (and added doc link)

File: eventarc/audit_iam/main.go (hunk 4 above)
Comment (muncus) [reply eventarc/audit_iam/main.go:64]: Thanks for mentioning this. i've linked the docs for the Request and Response objects.

File: eventarc/audit_iam/main.go
Comment (muncus) [reply eventarc/audit_iam/main.go:1]: Thanks for calling that out. I've added some "chunks" with separation.

File: eventarc/audit_iam/main.go (hunk 3 above)
Comment (telpirion) [reply eventarc/audit_iam/main.go]: Sounds good!

File: eventarc/audit_iam/main.go
Comment (telpirion) [reply eventarc/audit_iam/main.go:1]: Thank you!

**Review by telpirion — APPROVED** [review]

**Pull request merged**